biocrnpyler/dna_construct.py

```diff
@@ -397,12 +403,142 @@ def enumerate_components(self,previously_enumerated=None):
         combinatorial_components = self.combinatorial_enumeration()
 
         return combinatorial_components+new_constructs
-
-
+    @classmethod
+    def get_partstring(cls,part):
+        """a string name of a part including its name and direction (and not position)"""
+        orphan = part.get_orphan()
+        orphan.direction = None
+        orphan.position = None
+        curname = str(orphan)
+        curdir = part.direction
+        return curname+curdir
+    @classmethod
+    def get_partlist_hash(cls,partlist):
+        """creates a string containing the name and direction of all parts in a list of parts (but not their position)"""
+        partlist_str = '_'.join([str(a[0])+str(b) for a,b in zip(partlist,range(len(partlist)))])
+        return partlist_str
+    @classmethod
+    def hashless_reverse(cls,construct):
+        """create a reverse construct but don't calculate its hash (because that would make an infinite loop)"""
+        rev_con = [a.get_orphan() for a in construct]
+        for rev_part,origpart in zip(rev_con,construct):
+            rev_part.direction = {"forward":"reverse","reverse":"forward"}[origpart.direction]
+        rev_con = rev_con[::-1]
+        rev_con = Construct(rev_con,make_dirless_hash=False,circular = construct.circular)
+        return rev_con
+    @classmethod
+    def rotation_free_hash(cls,construct,return_rotation=False):
```

Comment:
`rotation_free_hash()` is always used with `return_rotation=True`. Can you remove the `False` branch?